src/arr/compiler/pyret.arr

```diff
@@ -145,31 +153,38 @@ fun main(args):
               display-progress: display-progress
             })
           failures = filter(CS.is-err, result.loadables)
-          when is-link(failures):
+          if is-link(failures):
             for each(f from failures) block:
               for lists.each(e from f.errors) block:
                 print-error(tostring(e))
                 print-error("\n")
               end
-              raise("There were compilation errors")
+              _ = print("There were compilation errors\n")
+              failure-code
             end
+          else:
+            success-code
           end
         else if r.has-key("run"):
-          CLI.run(r.get-value("run"), CS.default-compile-options.{
+          _ = CLI.run(r.get-value("run"), CS.default-compile-options.{
               standalone-file: standalone-file,
               compile-module: true,
               display-progress: display-progress,
               check-all: check-all
             })
+          success-code
         else:
-          print(C.usage-info(options).join-str("\n"))
-          raise("Unknown command line options")
+          _ = print(C.usage-info(options).join-str("\n"))
+          _ = print("Unknown command line options\n")
+          failure-code
         end
       end
     | arg-error(message, partial) =>
-      print(message + "\n")
-      print(C.usage-info(options).join-str("\n"))
+      _ = print(message + "\n")
+      _ = print(C.usage-info(options).join-str("\n"))
+      failure-code
   end
 end
 
-_ = main(C.args)
+exit-code = main(C.args)
+raise(ERR.exit-quiet(exit-code))
```

Comment:
This makes sense, though I wonder if `exit` and `exit-quiet` should be runtime methods that have the effect of throwing this special value.  I suggest this mainly because it may give us some more flexibility to update the implementation behind the interface later.